station/app/datasets/statistics.py

```diff
@@ -0,0 +1,183 @@
+import pandas as pd
+from typing import Optional
+from pandas.api.types import is_numeric_dtype
+from pandas import Series
+import plotly.express as px
+import plotly.io
+from plotly.graph_objects import Figure
+import json
+
+from station.app.schemas.datasets import DataSetStatistics, DataSetFigure
+from station.app.cache import redis_cache
+
+
+def get_dataset_statistics(dataframe: pd.DataFrame) -> Optional[DataSetStatistics]:
+    """
+    Computes statistical information of a dataset
+    :param dataframe: Dataset as dataframe object
+    :return: Dataset statistics
+    """
+    if not (isinstance(dataframe, pd.DataFrame)):
+        raise TypeError
+    shape = dataframe.shape
+    description = dataframe.describe(include='all')
+
+    n_items = shape[0]
+    n_features = shape[1]
+    columns_inf = get_column_information(dataframe, description)
+
+    schema_data = {
+        'n_items': n_items,
+        'n_features': n_features,
+        'column_information': columns_inf
+    }
+
+    statistics = DataSetStatistics(**schema_data)
+    return statistics
+
+
+def get_column_information(dataframe: pd.DataFrame, description: pd.DataFrame) -> dict:
+    """
+    Extract information out of dataframe and summarize it in a dictionary
+    :param dataframe: Dataframe to summarize
+    :param description: Dataframe description as dataframe object
+    :return: Dictionary with column information
+    """
+    columns_inf = []
+    columns = dataframe.columns.values.tolist()
+    for i in range(len(columns)):
+        title = columns[i]
+        count = description[title]["count"]
+        columns_inf.append({
+            'title': title
+        })
+        nan_count = dataframe[title].isna().sum()
+        if not(is_numeric_dtype(dataframe[title])):
```

Comment:
additional boolean check isnumeric(bool) = True, check for bool values 0,1 / True False